packages/scene-composer/src/components/three-fiber/AnimationComponent/AnimationComponent.tsx

```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line import/default
+import React, { Fragment, useEffect } from 'react';
+import * as THREE from 'three';
+import { useFrame } from '@react-three/fiber';
+
+import { useSceneComposerId } from '../../../common/sceneComposerIdContext';
+import { ISceneNodeInternal, useEditorState, useViewOptionState, IAnimationComponentInternal } from '../../../store';
+import { IAnimationComponent } from '../../../interfaces';
+
+const toggleIsAnimationPaused = (
+  object: THREE.Object3D,
+  AnimationMixer: THREE.AnimationMixer,
+  isPaused: boolean,
+  AnimationList: string[],
+) => {
+  try {
+    AnimationList.forEach(function (clip) {
+      const animation = THREE.AnimationClip.findByName(object.animations, clip);
+      AnimationMixer.clipAction(animation).paused = isPaused;
+    });
+  } catch (error) {
+    // console.error(error);
+  }
+};
+
+const list: any[] = [];
+interface AnimationComponentProps {
+  component: IAnimationComponent;
+  node: ISceneNodeInternal;
+}
+
+const AnimationComponent: React.FC<AnimationComponentProps> = ({ component, node }) => {
+  const sceneComposerId = useSceneComposerId();
+  const isGlobalAnimationPaused = !useViewOptionState(sceneComposerId).componentVisibilities['Animation'];
+  let obj;
+  const { getObject3DBySceneNodeRef } = useEditorState(sceneComposerId);
+  const object = getObject3DBySceneNodeRef(node.ref);
+  const currentAnimations: any[] = [];
+  if (object) {
```

Comment:
This is better as a useEffect and setState. As is it's getting set ever rerender which is potentially triggering the other useEffect that triggers on currentAnimations.